lib/SILOptimizer/Mandatory/AddressLowering.cpp

```diff
@@ -1290,6 +1306,9 @@ SILValue AddressLoweringState::getEarliestStoragePoint(SILValue userValue,
       // not allowed.  The projection will be inserted here.
       return SILValue();
     }
+    if (auto *loi = getLatestOpeningInst(pair)) {
+      return cast<SingleValueInstruction>(loi);
```

Comment:
Why do we only consider the first latest opening instruction? Can we way say something about the relationship between open archetypes based on the nature of use projections?

If we find a latest opening instruction, then why don't we also check dominance on the non-projection storage address the same way we do for the opening instruction?

Can we just lump all dependencies together and check dominance for each without caring about their relationship?

Reply:
> Why do we only consider the first latest opening instruction?

This was a false assumption I was making.  I added a test case demonstrating that.  It failed when considering only the first such instruction.

> Can we way say something about the relationship between open archetypes based on the nature of use projections?

We can say that they all dominate or are dominated by one another thanks to them all dominating the largest aggregate.  But we can't say anything about their relative ordering based on when they start appearing in the projection tower.

> If we find a latest opening instruction, then why don't we also check dominance on the non-projection storage address the same way we do for the opening instruction?

Fixed.

> Can we just lump all dependencies together and check dominance for each without caring about their relationship?

Yep, done.